source/globalCommands.py

```diff
@@ -4944,23 +4942,28 @@ def script_cycleSoundSplit(self, gesture: "inputCore.InputGesture") -> None:
 	)
 	def script_reportCaretLanguage(self, gesture: "inputCore.InputGesture"):
 		info = self._getTIAtCaret()
+		if info is None:
+			log.debugWarning("No caret")
+			return
 		info.expand(textInfos.UNIT_CHARACTER)
 		curLanguage = self._getCurrentLanguageForTextInfo(info)
-		langToReport = languageHandling.getLangToReport(curLanguage)
-		languageDescription = languageHandler.getLanguageDescription(langToReport)
+		if curLanguage is None:
+			# Translators: Reported when the language of the text at caret is not defined.
+			languageDescription = _("Not defined")
```

Comment:
```suggestion
			languageDescription = pgettext("reportLanguage", "Not defined")
```

Reply:
I would suggest "Unknown language" or "No set language" instead of "not defined". It seems a bit more user friendly. Thoughts @Qchristensen